pinot-controller/src/test/java/org/apache/pinot/controller/helix/PinotControllerModeTest.java

```diff
@@ -57,25 +61,42 @@ public void testHelixOnlyController()
   }
 
   @Test
-  public void testDualModeController()
-      throws Exception {
+  public void testDualModeController() throws Exception {
     config.setControllerMode(ControllerConf.ControllerMode.DUAL);
     config.setControllerPort(Integer.toString(Integer.parseInt(config.getControllerPort()) + controllerPortOffset++));
 
+    // Helix cluster will be set up when starting the first controller.
     startController(config);
     TestUtils.waitForCondition(aVoid -> _helixManager.isConnected(), TIMEOUT_IN_MS,
         "Failed to start " + config.getControllerMode() + " controller in " + TIMEOUT_IN_MS + "ms.");
     Assert.assertEquals(_controllerStarter.getControllerMode(), ControllerConf.ControllerMode.DUAL);
 
+    // Enable the lead controller resource.
```

Comment:
Do we need to explicitly enable the resource?

Reply:
By default it's disabled when creating a Helix cluster. In tests we'd like to explicitly enable it. That's why we need this line in the tests.